pulser/simulation/simulation.py

```diff
@@ -362,15 +368,11 @@ def evaluation_times(self, value: Union[str, ArrayLike, float]) -> None:
                     "Provided evaluation-time list contains "
                     "negative values."
                 )
-            # Ensure the list of times is sorted
-            eval_times = np.array(np.sort(value))
-            if t_min > 0:
-                eval_times = np.insert(eval_times, 0, 0.0)
-            if t_max < final_seq_time:
-                eval_times = np.append(eval_times, final_seq_time)
-
-            self._eval_times_array = eval_times
-            # always include initial and final times
+            # Ensure 0 and final time are included:
+            self._eval_times_array = np.union1d(
+                value, 
+                [0.0, self._tot_duration / 1000]
+                )
```

Comment:
Isn't this something that's always required? And if so, shouldn't we just do it at the end of the function for all cases?

Reply:
Yup, good idea.